kernel/src/protocols/errors.rs

```diff
@@ -75,6 +84,7 @@ impl From<SvsmError> for SvsmReqError {
             // to the guest as protocol-specific errors.
             SvsmError::SevSnp(e) => Self::protocol(e.ret()),
             SvsmError::InvalidAddress => Self::invalid_address(),
+            SvsmError::Apic(e) => Self::translate_apic_error(e),
```

Comment:
Any reason we are implementing `translate_apic_error()` instead of just inlining the conversion?

```suggestion
            SvsmError::Apic(e) => match e {
                ApicError::Emulation => Self::invalid_parameter(),
                ApicError::Registration => Self::protocol(SVSM_ERR_APIC_CANNOT_REGISTER),
            }
```

We can also do:

```suggestion
            SvsmError::Apic(ApicError::Emulation) => Self::invalid_parameter(),
            SvsmError::Apic(ApicError::Registration) => {
                Self::protocol(SVSM_ERR_APIC_CANNOT_REGISTER)
            }
```

Reply:
The thinking was that someday `ApicError` would grow to include new variants and that we didn't want `SvsmReqError::from` to be on a path to grow to an unmanageably large `match` statement.  Keeping different error types in different functions makes it a little easier to figure out how to organize the code as it grows, and the compiler will inline everything anyway (not that error handling needs to be optimized).  But I can go with your suggestion for now since the list is small.

Reply:
With my second suggestion, if new `ApicError` variants are added, they will result in the base case, so no further changes would be needed. With the first suggestion a compiler error will be emitted asking for all variants to be considered.